src/Utils.ts

```diff
@@ -21,34 +22,67 @@ export class Utils {
      * @returns {Boolean} the result of the check of the Editor mode
      */
     public static isEditMode(): boolean {
-        return PathUtils.getMetaPropertyValue(MetaProperty.WCM_MODE) === "edit" ? true : false;
+        const aemMode = PathUtils.getMetaPropertyValue(MetaProperty.WCM_MODE);
+        const param = new URL(document.location.href).searchParams.get(Constants.AEM_MODE);
```

Comment:
performance-wise the operation is not necessary if `aemMode` exists and has a value. Do we want to optimize this behavior?

Reply:
Server-Side Rendering may require some sanity check here 

_server-side applications can't reference browser-only global objects such as window, document, navigator, or location._

Reply:
Updated the implementation to work with SSR.
Please refer : 
https://github.com/adobe/aem-spa-page-model-manager/blob/issues/2/src/Utils.ts#L25
https://github.com/adobe/aem-spa-page-model-manager/blob/issues/2/src/ModelManager.ts#L148